maraboupy/MarabouNetworkNNet.py

```diff
@@ -13,40 +13,180 @@
+                                   outputMean=0, outputRange=1):
+        """Recompute the attributes of the network from the basic argumemts
+
+        Args:
+            weights (list of list of lists): Outer index corresponds to layer number
+            biases (list of lists): Outer index corresponds to layer number
+
+            inputMinimums (list of floats) Observed minimal values for the network
+            inputMaximums (list of floats) Observed maximal values for the network
+            inputMeans  (list of floats): Mean value for each input
+            inputRanges (list of floats): Range for each input
+            outputMean  (float): Mean value of outputs
+            outputRange (float): Range of output values
+
+        """
+
+        # Clearing the attributes that will be computed (clear() method from parent)
+        self.clear()
+
+        self.normalize = normalize
+
+        # Assert that the essential arguments are non-empty
+        assert weights
+        assert weights[0]
+        assert weights[0][0]
+        assert biases
+        assert biases[0]
+
+        # Compute network parameters that can be computed from the rest
+        numLayers = len(weights)
+        inputSize = len(weights[0][0])
+        outputSize = len(biases[-1])
+
+        # Initiate non-mandatory parameters
+        if not inputMinimums:
+            inputMinimums = [-sys.float_info.max]*inputSize
+        else:
+            inputMinimums = inputMinimums[:]
+        if not inputMaximums:
+            inputMaximums = [sys.float_info.max]*inputSize
+        else:
+            inputMaximums = inputMaximums[:]
+        if not inputMeans:
+            inputMeans = [0]*inputSize
+        else:
+            inputMeans = inputMeans[:]
+        if not inputRanges:
+            inputRanges = [1]*inputSize
+        else:
+            inputRanges = inputRanges[:]
+
+        # Find maximum size of any hidden layer
+        maxLayersize = inputSize
+        for b in biases:
+            if len(b) > maxLayersize:
+                maxLayersize = len(b)
+
+        # Create a list of layer Sizes
+        layerSizes = list()
+        layerSizes.append(inputSize)
+        for b in biases:
+            layerSizes.append(len(b))
+
+        self.numLayers = numLayers
+        self.layerSizes = layerSizes
+        self.inputSize = inputSize
+        self.outputSize = outputSize
+        self.maxLayersize = maxLayersize
+        self.inputMinimums = inputMinimums
+        self.inputMaximums = inputMaximums
+        self.inputMeans = inputMeans
+        self.inputRanges = inputRanges
+        self.outputMean = outputMean
+        self.outputRange = outputRange
+        self.weights = weights
+        self.biases = biases
+
+        self.computeNetworkAttributes()
+        # TODO (necessary? desirable?): create a new filename for the network
+
+    def computeNetworkAttributes(self):
```

Comment:
Short docstring needed here. We should add the ":meta private:" at the end of the docstring too, which makes the method a "private" method and hides the method from the API documentation.